pkg/preference/preference.go

```diff
@@ -0,0 +1,436 @@
+package preference
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo/pkg/util"
+)
+
+const (
+	globalConfigEnvName = "GLOBALODOCONFIG"
+	configFileName      = "odo-config.yaml"
+	//DefaultTimeout for openshift server connection check
+	DefaultTimeout = 1
+
+	// UpdateNotificationSetting is the name of the setting controlling update notification
+	UpdateNotificationSetting = "UpdateNotification"
+	// UpdateNotificationSettingDescription is human-readable description for the update notification setting
+	UpdateNotificationSettingDescription = "Controls if an update notification is shown or not (true or false)"
+	// NamePrefixSetting is the name of the setting controlling name prefix
+	NamePrefixSetting = "NamePrefix"
+	// NamePrefixSettingDescription is human-readable description for the name prefix setting
+	NamePrefixSettingDescription = "Default prefix is the current directory name. Use this value to set a default name prefix"
+	// TimeoutSetting is the name of the setting controlling timeout for connection check
+	TimeoutSetting = "Timeout"
+	// TimeoutSettingDescription is human-readable description for the timeout setting
+	TimeoutSettingDescription = "Timeout (in seconds) for OpenShift server connection check"
+)
+
+var (
+	// records information on supported parameters
+	supportedParameterDescriptions = map[string]string{
+		UpdateNotificationSetting: UpdateNotificationSettingDescription,
+		NamePrefixSetting:         NamePrefixSettingDescription,
+		TimeoutSetting:            TimeoutSettingDescription,
+	}
+
+	// set-like map to quickly check if a parameter is supported
+	lowerCaseParameters = util.GetLowerCaseParameters(GetSupportedParameters())
+)
+
+// GlobalConfigInfo wraps the global config and provides helpers to
+// serialize it.
+type GlobalConfigInfo struct {
```

Comment:
Wouldn't it make sense to rename this to `PreferencesInfo`?

Reply:
Least amount of change, we will again change all this hence not changed it

Reply:
Why we will change all this again?

Reply:
I was going for least amount of change, I can change this if you feel?

Reply:
@kadel we don't have to but if global config is named preferences now, it makes sense to change the associated types as well…

Reply:
I would prefer to change it now. It doesn't really make sense to rename it only in some places and leave the rest. We we should remove all references to global config and use just preference

Reply:
renamed to prefrenceInfo